src/main/java/ch/njol/skript/expressions/ExprCoordinate.java

```diff
@@ -76,47 +78,58 @@ public Class<? extends Number> getReturnType() {
 	@Override
 	@Nullable
 	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if ((mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) && getExpr().isSingle() && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Location.class))
-			return new Class[] {Number.class};
+		if (getExpr() instanceof Location) {
+			if ((mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) && getExpr().isSingle() && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Location.class))
+				return new Class[] {Number.class};
+		} else if (getExpr() instanceof Chunk) {
+			if ((mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) && getExpr().isSingle() && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Chunk.class))
+				return new Class[] {Number.class};
+		}
```

Comment:
Avoid duplicated code
```suggestion
		if ((mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) && getExpr().isSingle()) {
			if (
				getExpr() instanceof Location && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Location.class) || 
				getExpr() instanceof Chunk && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Chunk.class)
			) {
				return new Class[] {Number.class};
			}
		}
```

Reply:
While looking at `change` method I see no support for chunks? seems like Chunks can have their X/Z coords changed therefore, this whole change to acceptChange method is not needed, you can just check if `getExpr() instanceof Chunk` then throw a Skript#error saying something like `"Chunks can't have their X/Y coords changed"` and then the whole change to `change` method is not needed as well